kscale/store/urdf.py

```diff
@@ -1 +1,116 @@
 """Utility functions for managing artifacts in the K-Scale store."""
+
+import asyncio
+import logging
+import os
+import sys
+import tarfile
+from typing import Sequence
+
+import httpx
+import requests
+
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+CACHE_DIR = os.path.expanduser("~/.cache/kscale")
+
+
+def fetch_urdf_info(listing_id: str) -> dict:
+    url = f"https://api.kscale.store/urdf/info/{listing_id}"
+    response = requests.get(url)
+    response.raise_for_status()
+    return response.json()
+
+
+def download_artifact(artifact_url: str, cache_dir: str) -> str:
+    os.makedirs(cache_dir, exist_ok=True)
+    filename = os.path.join(cache_dir, artifact_url.split("/")[-1])
+
+    if not os.path.exists(filename):
+        logger.info(f"Downloading artifact from {artifact_url}")
```

Comment:
nit: don't use format strings in logger messages

```suggestion
        logger.info("Downloading artifact from %s", artifact_url)
```